src/optimize.jl

```diff
@@ -203,3 +203,46 @@ function optimize!(θ::AbstractVector, phases::AbstractVector{<:CrystalPhase},
 	@. θ = exp(log_θ) # transform back
 	return θ
 end
+
+# optimization based on SaddleFreeNewton method
+function newton!(θ::AbstractVector, phases::AbstractVector{<:CrystalPhase},
```

Comment:
Note: I haven't tested this code, because I did not get the optimize.jl test to work. Idea: can you add just the synthetic data to the /test folder, so that we can execute the tests on any machine?